course/src/main/java/in/testpress/course/util/ManageCourseStates.java

```diff
@@ -0,0 +1,81 @@
+package in.testpress.course.util;
+
+import android.util.Log;
+
+import java.util.List;
+
+import in.testpress.course.enums.CourseType;
+import in.testpress.models.greendao.Course;
+import in.testpress.models.greendao.CourseDao;
+
+public class ManageCourseStates {
+    private CourseType type;
+    private CourseDao courseDao;
+    private List<Course> courses;
+
+    public ManageCourseStates(CourseType type, CourseDao courseDao, List<Course> courses) {
+        this.type = type;
+        this.courseDao = courseDao;
+        this.courses = courses;
+    }
+
+    private void updateState(Course course, boolean state) {
```

Comment:
No need two flags. Just my course is enought. Also this will fail on second page.